prov/rxm/src/rxm_ep.c

```diff
@@ -2382,8 +2382,10 @@ static int rxm_listener_open(struct rxm_ep *rxm_ep)
 	}
 
 	ret = fi_control(&rxm_ep->msg_eq->fid, FI_GETWAIT, &rxm_ep->msg_eq_fd);
-	if (ret)
-		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "Unable to get MSG EQ fd\n");
+	if (ret) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "unable to get MSG EQ fd\n");
+		goto err;
```

Comment:
I believe this has always been failing on macOS. But now its visible.

Reply:
Is this something that can be fixed in the tcp provider or util code?  Is this from epoll emulation on OS X?

Reply:
Looking at the rxm code, I don't see a strong reason why the msg_eq_fd is acquired here.  The msg_eq_fd is needed when binding a CQ that is associated with a wait set to the EP, or when doing something with atomic progress (rxm_conn_atomic_progress_eq_cq()).  The latter case tries to get the msg_eq_fd independently.

If we're not dealing with either of those cases, then the msg_eq_fd isn't needed.  The fi_control() call can be moved out of here and into rxm_ep_wait_fd_add.  That way, we only try to access it if it's actually needed.

The msg_cq_fd has a similar access pattern.  It is needed with atomic progress (rxm_conn_atomic_progress_eq_cq()), but unlike the msg_eq_fd, that function does not call fi_control() to get it and assumes it's already available.  It is also needed by rxm_ep_wait_fd_add().

I suggest modifying rxm to have a single call that retrieves both the msg_eq_fd and msg_cq_fd, which is called from rxm_conn_atomic_progress_eq_cq() and rxm_ep_wait_fd_add().  I don't think we to store either fd with the rxm_ep.  The fd's could be obtained and used directly in the two functions that use them.

Reply:
FI_GETWAIT implies wait object of type FI_WAIT_FD to be return by util code. But providing the emulated epoll fd won't help because calling system level poll calls on it won't detect the changes on the fds that are represented by emulated epoll fd. 

So may be using kqueue in this case would help for macOS.

Reply:
Thanks @vkrishna for checking this. I'll update the code based on @shefty 's suggestion.

Are tests that use CQ wait objects excluded for tcp provider on Mac OS?

Reply:
Yes